installer2/WindowsInfo.java

```diff
@@ -1,47 +1,67 @@
-import java.io.*;
-
-public class WindowsInfo
-{
-    public static void main(String[] args)
-    {
-        String path = System.getenv("ProgramFiles(x86)");
-        File directory = null;
-        if (path != null)
-        {
-            directory = getInstallDir(path);
+import VDF.VDF;
+import com.sun.deploy.util.WinRegistry;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+Made using the SteamVDF library
+https://github.com/StidOfficial/SteamVDF
+ **/
+
+public class WindowsInfo {
+    public static void main(String[] args) {
+        String[] files = new String[]
+                {
+                        "Terraria.exe",
+                        "ModCompile/tModLoaderMac.exe",
+                        "tModLoaderServer.exe",
+                        "ModCompile/FNA.dll",
+                        "ModCompile/Microsoft.CodeAnalysis.dll",
+                        "ModCompile/Microsoft.CodeAnalysis.CSharp.dll",
+                        "ModCompile/Mono.Cecil.Pdb.dll",
+                        "ModCompile/System.Reflection.Metadata.dll",
+                        "ModCompile/RoslynWrapper.dll",
+                        "start-tModLoaderServer.bat",
+                        "start-tModLoaderServer-steam-friends.bat",
+                        "start-tModLoaderServer-steam-private.bat"
+                };
+
+        String steamPath = WinRegistry.getString(WinRegistry.HKEY_CURRENT_USER, "SOFTWARE\\Valve\\Steam", "SteamPath");
+
+        File steamInstall = new File(steamPath + "/steamapps/appmanifest_105600.acf");
+        if (steamInstall.exists()) {
+            Installer.tryInstall(files, new File(steamPath + "/steamapps/common/Terraria"));
+            return;
         }
-        if (directory == null || !directory.exists())
-        {
-            path = System.getenv("ProgramFiles");
-            if (path != null)
-            {
-                directory = getInstallDir(path);
+
+        File InputVDFFile = new File(steamPath + "/config/config.vdf");
+        VDF InputVDF = new VDF(InputVDFFile);
+        BufferedReader buffer = new BufferedReader(new FileReader(steamPath + "/config/config.vdf"));
+
+        String line = null;
+        while ((line = buffer.readLine()) != null) {
+            Pattern pattern = Pattern.compile("\"(.*?)\"");
+            Matcher matcher = pattern.matcher(line);
+
+            while (matcher.find()) {
+                String key = matcher.group(1);
+                if (key.contains("BaseInstallFolder")) {
```

Comment:
in general I'd recommend inverting conditions and using continue to reduce nesting, but it's fine here